File: challenge-325/pokgopun/go/ch-1.go
//# https://theweeklychallenge.org/blog/perl-weekly-challenge-325/
/*#

Task 1: Consecutive One

Submitted by: [43]Mohammad Sajid Anwar
     __________________________________________________________________

   You are given a binary array containing only 0 or/and 1.

   Write a script to find out the maximum consecutive 1 in the given
   array.

Example 1

Input: @binary = (0, 1, 1, 0, 1, 1, 1)
Output: 3

Example 2

Input: @binary = (0, 0, 0, 0)
Output: 0

Example 3

Input: @binary = (1, 0, 1, 0, 1, 1)
Output: 2

Task 2: Final Price
#*/
//# solution by pokgopun@gmail.com

package main

import (
	"io"
	"iter"
	"os"
	"slices"

	"github.com/google/go-cmp/cmp"
)

func co(ints iter.Seq[int]) int {
	var c, mx, v int
	for v = range ints {
		if v == 0 {
			if c == 0 {
				continue
			}
			if c > mx {
				//fmt.Printf("mx = %d\n", c)
				mx = c
			}
			c = 0
		} else {
			c++
		}
	}
	if v > 0 {
		//fmt.Println("final max update")
		if c > mx {
			//fmt.Printf("mx = %d\n", c)
			mx = c
		}
	}
	return mx
}

func main() {
	for _, data := range []struct {
		input  iter.Seq[int]
		output int
	}{
		{slices.Values([]int{0, 1, 1, 0, 1, 1, 1}), 3},
		{slices.Values([]int{0, 0, 0, 0}), 0},
		{slices.Values([]int{1, 0, 1, 0, 1, 1}), 2},
		{slices.Values([]int{1, 0, 1, 1, 0, 1, 1}), 2},
		{slices.Values([]int{1, 0, 1, 1, 0, 0, 1, 1}), 2},
		{slices.Values([]int{0, 1, 1, 1, 0, 1, 1, 1}), 3},
		{slices.Values([]int{1, 1, 1, 1, 0, 1, 1, 1, 0}), 4},
		{slices.Values([]int{0, 1, 1, 0, 0, 1, 1, 1, 0}), 3},
	} {
		//fmt.Println(data)
		io.WriteString(os.Stdout, cmp.Diff(co(data.input), data.output)) // blank if ok, otherwise show the difference
	}
}

File: challenge-326/pokgopun/python/ch-1.py
### https://theweeklychallenge.org/blog/perl-weekly-challenge-326/
"""

Task 1: Day of the Year

Submitted by: [43]Mohammad Sajid Anwar
     __________________________________________________________________

   You are given a date in the format YYYY-MM-DD.

   Write a script to find day number of the year that the given date
   represent.

Example 1

Input: $date = '2025-02-02'
Output: 33

The 2nd Feb, 2025 is the 33rd day of the year.

Example 2

Input: $date = '2025-04-10'
Output: 100

Example 3

Input: $date = '2025-09-07'
Output: 250

Task 2: Decompressed List
"""
### solution by pokgopun@gmail.com

from time import strptime

def doy(date: str) -> int:
    return strptime(date, "%Y-%m-%d").tm_yday

import unittest

class TestDoy(unittest.TestCase):
    def test(self):
        for inpt, otpt in {
                '2025-02-02': 33,
                '2025-04-10': 100,
                '2025-09-07': 250,
                }.items():
            self.assertEqual(doy(inpt),otpt)

unittest.main()
